### DISCUSSION TITLE
2021.10.04 회의록

### DISCUSSION THREAD

**Discussion by kwakjeeyoon** [boostcampaitech2/object-detection-level2-cv-06#53] [General]
# 회의록
- 화요일 아침에 토의하면 멘토링때 여쭤볼 내용을 정리하기에 시간이 부족할것 같아 회의를 소집함
- 작업을 할때 프로젝트 관점에서 생각을 했으면 좋겠다. - 코드의 재사용성이 부족하다.
- **validation set**을 나눠서 실험을 해봐야 한다 (실험 초반에 세팅을 해야 한다.)
    - validation set 을 고정해서 진행
    - k-fold 방법은 적용 X (mmdetection에서 제공하지 않는걸로 암)
- train/val 데이터셋을 다각화 - augmentation으로 원래 데이터와 augmentation을 진행한 데이터를 train/val로 데이터셋으로 하여 진행하는 방법도 좋을 것 같다.
    - → 단순 분류에는 적용이 가능하지만 detection 문제에는 알맞지 않는 방법인것 같다.
    - → train/val이 augemntation이라도 원본이 같은 데이터를 공유하면 val의 데이터셋으로 적합하지 않다.
    - 모자이크 augmentation 적용을 하면 어떨까
    - 멘토링때 여쭤보도록 하자
    - 8:2로 최적의 epoch과 shceduler를 찾고 10:0으로 제출하는 접근도 있다.
- detr와 비교해서 underfit 되었다는 말이 어떤 의미인지 말씀해주실수 있으신가요
    - train/val을 같이 써서 overfiting 정도를 파악하지 못한다.
- 학습 결과를 한눈에 볼 수 있는 테이블을 만들 필요가 있다.

### 요약

**범용성 있는 코드로 만들기!!**

- 결과를 한눈에 볼 수 있는 테이블 생성
- validation set 을 추가
    - 정해진 비율로 validation set을 뽑고 랜덤하게 100개 validation set을 뽑아 리더보드와 성능이 가장 비슷한 것을 validation set으로 고정한다.

⇒ 서지유, 오재환, 전경재, 이준혁 캠퍼님

- early_stop 기능을 추가
    - best_model을 정하는 걸로 구현할 수 있다. ← 이걸로 진행
- 추가 모델
    - detectron 과 앙상블하면 모델 성능이 올라간다고 한다
    - 모든 모델을 돌리는 것이 의미가 있을까 - 가설을 세우고 그것을 증명하는 방식으로 진행하는 것이 좋을듯 하다.

⇒ 강지우, 곽지윤, 송나은 캠퍼님

- Adam으로 진행

## validation 데이터 팀(10-5일까지)

- 1 stage, 2 stage , transformer 3 모델로 valid 실험
    
    1 stage : SSD (재환)
    
    2 stage : Faster_rcnn (경재)
    
    Multi stage : Hybrid Task Cascade (준혁)
    
    최종 검증용→ tf : deformable_detr 
    
    [(train, valid)→(ssd, frcnn)] xN → 최종 검증 tf
    
    샘플링 방법
    
    - 랜덤(9:1) → (sample 500개)
    - 시간되는 대로 많이
    - 랜덤샘플→ EDA→test
        
        
    
    통계 낼 모델(3+1개)(1-stage, 2-stage, Multi-stage, tf)
    
    Image 512x512로 진행 / optimizer = Adam / seed 1004 고정
    
    transformer 
    
    img_scale=[(480, 1024), (512, 1024), (544, 1024), (576, 1024),(608, 1024), (640, 1024), (672, 1024), (704, 1024),(736, 1024), (768, 1024), (800, 1024)],
    
    → 각각 1/2
    
    추가 커스텀(작은 bbox → outlier로 쳐내야함)
    

## model팀 (10월 6일 까지)

soft teacher → 곽지윤

- [https://github.com/microsoft/SoftTeacher](https://github.com/microsoft/SoftTeacher)
- mmdetection 구현

cbnet → 송나은

- [https://paperswithcode.com/paper/cbnetv2-a-composite-backbone-network](https://paperswithcode.com/paper/cbnetv2-a-composite-backbone-network)
- mmdetection 구현

copy paste → 강지우

- [https://github.com/conradry/copy-paste-aug](https://github.com/conradry/copy-paste-aug)

![image](https://user-images.githubusercontent.com/55044675/135805407-1b47fb3a-3f28-4f88-a28f-24c448a595ed.png)

queryInst(순위낮음, detr 변형 모델)

- [https://github.com/hustvl/QueryInst/tree/main/configs/queryinst](https://github.com/hustvl/QueryInst/tree/main/configs/queryinst)
- [https://github.com/megvii-research/SOLQ](https://github.com/megvii-research/SOLQ)
- mmdetection에 구현되어있지 않음
